packages/util/src/util.ts

```diff
@@ -0,0 +1,12 @@
+export type PromiseOrValue<T> = Promise<T> | T;
+
+export const resolveObjectOfPromises = async <T>(obj: {
+  [k: string]: PromiseOrValue<T>;
+}): Promise<{ [k: string]: T }> =>
+  Object.fromEntries(
+    await Promise.all(
+      Object.entries(obj).map(([key, promise]) =>
+        promise instanceof Promise ? promise.then((value) => [key, value]) : [key, promise]
```

Comment:
`instanceof Promise` can be brittle, there are many libraries that use their own Promise-like types, polyfills etc.
```suggestion
        Promise.resolve(promise).then((value) => [key, value])
```

Nice util btw :1st_place_medal:

Reply:
Nice! Thank you

Reply:
60b094607 should solve this